scala/private/macros/setup_scala_toolchain.bzl

```diff
@@ -0,0 +1,74 @@
+load("//scala:scala_toolchain.bzl", "scala_toolchain")
+load("//scala:providers.bzl", "declare_deps_provider")
+
+def setup_scala_toolchain(
+        name,
+        scala_xml_deps,
```

Comment:
nit: why some attribute names ends with deps and others with classpath? What's the distinction?

Reply:
`classpath` is what historically was used on the toolchain, other names came when dep providers were introduced. How I see it: `classpath` is implicit deps on the classpath, and `deps` is tool related deps. But this naming assumption does not even hold with the current codebase (eg. compiler is tool, but has classpath naming). Not sure what to do about it. It's a good time to improve those names at least on the macro side. Naming suggestions are more than welcome. @simuons, WDYT?

Reply:
I don't really have good suggestion. I wasn't aware of this distinction about `classpath` and `tools`. My initial thought was "all those deps go to the classpath at some point so why the distinction". The question probably is wether users should set 'internal' deps for things like `//scala/support:test_reporter` but this is out of scope for this PR. So I'm ok with state as it is now.

Reply:
After thinking a bit more, I'll change it to use default providers for tools deps. This should give us more room to do future refactorings. Eg. move scala_xml dep to testing toolchain.